src/models/tiles/geometry/geometry-content.ts

```diff
@@ -688,8 +692,20 @@ export const GeometryContentModel = GeometryBaseContentModel
       const result = syncChange(board, change);
       const point = isPoint(result) ? result : undefined;
 
-      if (point && polygonId) {
-        appendPhantomPointToPolygon(board, polygonId);
+      if (point && self.activePolygonId && restoring) {
+        appendPhantomPointToPolygon(board, self.activePolygonId);
+      }
+      if (point && self.activeCircleId && restoring) {
```

Comment:
Could write this a bit more succinctly as:
```suggestion
      if (point && restoring) {
        if (self.activePolygonId) {
          appendPhantomPointToPolygon(board, self.activePolygonId);
        }
        if (self.activeCircleId) {
          ...
        }
      }
```